src/daily_output.ts

```diff
@@ -318,6 +315,26 @@ export class DailyOutput {
 		this.latestTimestamp = Math.max(this.latestTimestamp, node.block.timestamp);
 	};
 
+	private isNewESTDate(prev: Date, curr: Date): boolean {
+		const isNewDay = this.dateToEST(prev).getDate() !== this.dateToEST(curr).getDate();
+		return isNewDay;
+	}
+
+	private isNewESTWeek(prev: Date, curr: Date): boolean {
+		const isNewDay = this.dateToEST(prev).getDay() !== this.dateToEST(curr).getDay();
+		return isNewDay;
+	}
+
+	private dateToEST(d: Date): Date {
+		const date = new Date(d.getTime());
+		const offset = date.getTimezoneOffset(); // getting offset to make time in gmt+0 zone (UTC) (for gmt+5 offset comes as -300 minutes)
+		date.setMinutes(date.getMinutes() + offset); // date now in UTC time
+
+		const easternTimeOffset = -240; // for dayLight saving, Eastern time become 4 hours behind UTC thats why its offset is -4x60 = -240 minutes. So when Day light is not active the offset will be -300
```

Comment:
Is it possible to detect daylight saving here? Having to remember to manually change this is probably tech debt

Reply:
Although, when they remove daylight savings or make it permanent we'd have to change this anyway.